src/main/scala/zio/rocksdb/Deserializer.scala

```diff
@@ -0,0 +1,37 @@
+package zio.rocksdb
+
+import zio.ZIO
+
+trait Deserializer[-R, +A] { self =>
+  def decode(bytes: Bytes): ZIO[R, DeserializeError, Result[A]]
+
+  final def map[B, A1 >: A](f: A1 => B): Deserializer[R, B] =
```

Comment:
Is `A1` needed here? It's usually not a problematic position for covariant type variables

Reply:
Yup, I thought so but you are right.